drivers/openstack/client.go

```diff
@@ -439,12 +444,26 @@ func (c *GenericClient) Authenticate(d *Driver) error {
 
 	provider.UserAgent.Prepend(fmt.Sprintf("docker-machine/v%s", version.VERSION))
 
+	// Configure custom TLS settings.
 	if d.Insecure {
-		// Configure custom TLS settings.
-		config := &tls.Config{InsecureSkipVerify: true}
-		transport := &http.Transport{TLSClientConfig: config}
-		provider.HTTPClient.Transport = transport
+		config = &tls.Config{InsecureSkipVerify: true}
+	} else {
+		flCa := filepath.Join(utils.GetMachineCertDir(), "ca.crt")
```

Comment:
Why the default value `ca.crt` ? With your code the use of the certificate becomes mandatory. It's a mistake. In my opinion the code should look something like

```ruby
if d.Insecure {
	config = &tls.Config{InsecureSkipVerify: true}
}

if d.TlsCACert != "" {
	flCa := filepath.Join(utils.GetMachineCertDir(), d.TlsCACert)
	certPool := x509.NewCertPool()
	file, err := ioutil.ReadFile(flCa)
	if err != nil {
		return err
	}
	certPool.AppendCertsFromPEM(file)
	config = &tls.Config{
		RootCAs: certPool,
		InsecureSkipVerify: false,
	}
}

transport := &http.Transport{TLSClientConfig: config}
provider.HTTPClient.Transport = transport
```